endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints.go

```diff
@@ -103,22 +103,24 @@ func removeOldEndpoints(ctx context.Context, store *postgresql.Store, updateTime
 			log.Warn(err)
 			continue
 		}
-		existingEndpoint, err := store.GetFHIREndpointInfoUsingURL(ctx, endpoint.URL)
+		existingEndpointList, err := store.GetFHIREndpointInfosUsingURL(ctx, endpoint.URL)
 		if err == sql.ErrNoRows {
 			log.Warn(err)
 			continue
 		} else {
-			endpointList, err := store.GetFHIREndpointUsingURL(ctx, endpoint.URL)
-			if err != nil {
-				log.Warn(err)
-				continue
-			}
-			if len(endpointList) == 0 {
-				err = store.DeleteFHIREndpointInfo(ctx, existingEndpoint)
+			for _, existingEndpoint := range existingEndpointList {
+				endpointList, err := store.GetFHIREndpointUsingURL(ctx, endpoint.URL)
 				if err != nil {
 					log.Warn(err)
 					continue
 				}
+				if len(endpointList) == 0 {
```

Comment:
I think you could rewrite this so that instead of looping on line 11 and calling GetFhirEndpointUsingURL every time, you could just call it once using the endpoint.URL, and then if it doesn't exist anymore, put the for loop in this if check instead and delete all of the endpoints that way?